nornir/core/inventory.py

```diff
@@ -336,39 +340,43 @@ def open_connection(
         Returns:
             An already established connection
         """
-        if connection in self.connections:
-            raise ConnectionAlreadyOpen(connection)
+        conn_name = connection
+        existing_conn = self.connections.get(conn_name)
+        if existing_conn is not None and existing_conn is not UNESTABLISHED_CONNECTION:
+            raise ConnectionAlreadyOpen(conn_name)
 
-        self.connections[connection] = self.connections.get_plugin(connection)()
+        plugin = self.connections.get_plugin(conn_name)
+        conn_obj = plugin()
         if default_to_host_attributes:
-            conn_params = self.get_connection_parameters(connection)
-            self.connections[connection].open(
-                hostname=hostname if hostname is not None else conn_params.hostname,
-                username=username if username is not None else conn_params.username,
-                password=password if password is not None else conn_params.password,
-                port=port if port is not None else conn_params.port,
-                platform=platform if platform is not None else conn_params.platform,
-                extras=extras if extras is not None else conn_params.extras,
-                configuration=configuration,
-            )
-        else:
-            self.connections[connection].open(
-                hostname=hostname,
-                username=username,
-                password=password,
-                port=port,
-                platform=platform,
-                extras=extras,
-                configuration=configuration,
-            )
-        return self.connections[connection]
+            conn_params = self.get_connection_parameters(conn_name)
+            hostname = hostname if hostname is not None else conn_params.hostname
+            username = username if username is not None else conn_params.username
+            password = password if password is not None else conn_params.password
+            port = port if port is not None else conn_params.port
+            platform = platform if platform is not None else conn_params.platform
+            extras = extras if extras is not None else conn_params.extras
+
+        conn_obj.open(
+            hostname=hostname,
+            username=username,
+            password=password,
+            port=port,
+            platform=platform,
+            extras=extras,
+            configuration=configuration,
+        )
+        self.connections[conn_name] = conn_obj
+        return connection
 
     def close_connection(self, connection: str) -> None:
         """ Close the connection"""
-        if connection not in self.connections:
-            raise ConnectionNotOpen(connection)
+        conn_name = connection
+        if conn_name not in self.connections:
+            raise ConnectionNotOpen(conn_name)
 
-        self.connections.pop(connection).close()
+        connection = self.connections.pop(conn_name)
```

Comment:
this should actually trip mypy. You are storing a `type` of some sort in a variable that was previously defined as a `str` so maybe it's better to skip the `conn_name` and rename this to `conn = self.connections.pop(conn_name)`